### DISCUSSION TITLE
Fedora 36 Traefik gateway error

### DISCUSSION THREAD

**Discussion by Harnish** [k3s-io/k3s#6268] [General]
**Environmental Info:**
K3s Version: k3s version v1.24.6+k3s1 (a8e0c66d)

Node(s) CPU architecture, OS, and Version: 
5.19.14-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Oct 5 21:31:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
1 server Oracle Virtualbox, 2 CPUS, 4 gigs of ram 

**Describe the bug:**
Following a very basic setup, traefik returns a 502 bad gateway.  Steps below.  When run on Ubuntu 22.04.1 the same process works.  

**Steps To Reproduce:**

- Install Oracle Virtual Box 7.0.
- Create a Fedora 36 vm using the Fedora 36 server (https://download.fedoraproject.org/pub/fedora/linux/releases/36/Server/x86_64/iso/Fedora-Server-dvd-x86_64-36-1.5.iso)
- Set the Network to be bridged.
- Do a Standard install of Fedora 36.
- Once installed and booted.  Login and do dnf update -y
- Reboot
- Installed K3s: curl -sfL https://get.k3s.io | sh -
- setup env:  mkdir .kube; cd .kube; sudo cp /etc/rancher/k3s/k3s.yaml config; chown <user> config; cd ..; echo "export KUBECONFIG=~/.kube/config" >>~/.bashrc; source ~/.bashrc
- kube create namespace dev
- kube apply -f whoami.deployment.yaml
- kube apply -f whoami.service.yaml
- kube apply -f whoami.ingress.yaml
- create a dns record for whoami.k3s.local to point to the ip of the vm.
- using a webbrowser: http://whoami.k3s.local

**Expected behavior:**

I expect a response like: 
Hostname: whoami-74db99b4d6-cnz8l
IP: 127.0.0.1
IP: ::1
IP: 10.42.0.9
IP: fe80::ec20:cfff:feba:6047
RemoteAddr: 10.42.0.7:50596
GET / HTTP/1.1
Host: whoami.k3s4.local
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:105.0) Gecko/20100101 Firefox/105.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.5
Sec-Gpc: 1
Upgrade-Insecure-Requests: 1
X-Forwarded-For: 10.42.0.1
X-Forwarded-Host: whoami.k3s4.joe.beardedgeek.org
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: traefik-7cd4fcff68-h9ckz
X-Real-Ip: 10.42.0.1

**Actual behavior:**

Bad Gateway

**Additional context / logs:**
This exact same process works just fine using Ubuntu 22.04.1.  

I couldn't upload yaml extensions so I added txt.  
[whoami.ingress.yaml.txt](https://github.com/k3s-io/k3s/files/9777638/whoami.ingress.yaml.txt)
[whoami.service.yaml.txt](https://github.com/k3s-io/k3s/files/9777639/whoami.service.yaml.txt)
[whoami.deployment.yaml.txt](https://github.com/k3s-io/k3s/files/9777641/whoami.deployment.yaml.txt)

**Comment by brandond** [comment]
Your request says:

>```
>GET / HTTP/1.1
>Host: whoami.k3s4.local
>```

But your ingress has: `host: whoami.k3s.local`. Are you sure you're testing your ingress properly?

Either way, I would probably take this question to the Traefik project. We bundle an unmodified version of Traefik, and aren't experts in its use.

**Reply by Harnish** [reply]
The request was proper.  k3s was my Fedora vm and the k3s4 is the Ubuntu vm.  Further investigation looks like it is a networking problem internal to kubernetes.  Traefik is timing out trying to get to the backend pod on the specified port.

**Comment by Harnish** [comment]
I assumed the problem was with selinux, so I set it to permissive and reran the tests with the same outcome.

**Comment by Harnish** [comment]
Found the fix in your documentation:

https://docs.k3s.io/advanced#additional-preparation-for-red-hatcentos-enterprise-linux

Once I did this Fedora 36 worked just fine.
